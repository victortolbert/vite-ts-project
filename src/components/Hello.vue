<script setup lang="ts">
const times = ref(2)
const props = defineProps<{ count: number }>()
const result = computed(() => props.count * times.value)
defineExpose(props)
</script>

<template>
  <Foo>
    <div>{{ count }} x {{ times }} = {{ result }}</div>
    <button @click="times += 1">x1</button>
  </Foo>
</template>