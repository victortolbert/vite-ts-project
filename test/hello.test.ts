import { mount } from '@vue/test-utils'
import Hello from '../src/components/Hello.vue'

test.skip('mount component', async () => {
  expect(Hello).toBeTruthy()

  const wrapper = mount(Hello, {
    propsData: {
      count: 4,
    },
  })

  expect(wrapper.text()).toContain('4 x 2 = 8')
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('button').trigger('click')
  await nextTick()

  expect(wrapper.text()).toContain('4 x 3 = 12')

  await wrapper.get('button').trigger('click')
  await wrapper.get('button').trigger('click')

  expect(wrapper.text()).toContain('4 x 5 = 20')
  expect(wrapper.html()).toMatchSnapshot()
})
