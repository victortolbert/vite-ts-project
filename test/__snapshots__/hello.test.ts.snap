// Vitest Snapshot v1

exports[`mount component 1`] = `
"<foo>
  <div>4 x 2 = 8</div><button> x1 </button>
</foo>"
`;
